File: app2/src/main/java/com/sample/app2/MainActivity.java
package com.sample.app2;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import android.os.Bundle;

import com.sample.app2.databinding.ActivityMainBinding;
import com.sample.lib2.Lib2Utils;

public class MainActivity extends AppCompatActivity {
    ActivityMainBinding binding;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this,R.layout.activity_main);
        StringBuffer buffer = new StringBuffer();
        buffer.append("\n\n")
                .append(Lib2Utils.getName());
        binding.text.append(buffer);
    }
}

File: app/src/main/java/com/example/codemultiplechannel/MainActivity.kt
package com.example.codemultiplechannel

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import com.example.codemultiplechannel.databinding.ActivityMainBinding
import com.sample.lib1.Lib1Utils
import com.sample.lib2.Lib2Utils
import com.sample.lib2.Lib2UtilsStringGenerater
import com.sample.lib3.Lib3Utils

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = DataBindingUtil
            .setContentView<ActivityMainBinding>(this,R.layout.activity_main)
        binding.tv.run {
            append("\n\n")
            append(Utils.getStr())
            append("\n\n")
            append(Lib1Utils.getName())
            append("\n\n")
            append(Lib2UtilsStringGenerater.getName())
            append("\n\n")
            append(Lib3Utils.getName())
            append("\n\n")
        }


    }
}